src/languages/scala.js

```diff
@@ -151,11 +174,25 @@ export default function(hljs) {
     beginScope: { 2: "keyword", }
   };
 
+  const SOFT_MODIFIERS = [ 'infix', 'inline', 'opaque', 'open', 'transparent' ];
+  const SOFT_MODIFIER_RE = regex.either(...SOFT_MODIFIERS);
+
+  const SOFT_MODIFIER_MODE = {
+    begin: regex.concat(
+      /(^|;)\s*/,
```

Comment:
Is this truly necessary with all the context provided by the look-ahead? Do you have an example of a false positive if this was removed? 

Generally we don't try and match on "prior" content as we may want to match that in the future such as the `;` should long-term be flagged as punctuation, etc... and it's harder to do that with rules like this.

Reply:
Sure, false positive cases without `/(^|;)\s*/`:

### case 1
`notinline given Int = 9`

```
      -not<span class="hljs-keyword">inline</span> <span class="hljs-keyword">given</span> <span class="hljs-type">Int</span> = <span class="hljs-number">9</span>
      +notinline <span class="hljs-keyword">given</span> <span class="hljs-type">Int</span> = <span class="hljs-number">9</span>
```

### case 2

`a.infix == b.infix`
```
      -a.infix == b.<span class="hljs-keyword">infix</span>
      +a.infix == b.infix
```

I understand what you're worried about and will be happy to get some advice to improve this.

Reply:
BTW, thanks for the thorough review so far!

Reply:
Typically case one you can use a `\b` at the beginning of the expression.

Case two what we usually do is match `.thing` separately with another rule (these are `property` accesses) and that avoids them being mismatched as keywords.

Reply:
Thanks for the comment.

For case 1, if I replace it with `\b` the following false positives arise:
1. `val transparent = 43` 
```
      -<span class="hljs-keyword">val</span> <span class="hljs-keyword">transparent</span> = <span class="hljs-number">43</span>
      +<span class="hljs-keyword">val</span> transparent = <span class="hljs-number">43</span>
```

2. `c opaque d`
```
      -c <span class="hljs-keyword">opaque</span> d
      +c opaque d
```

For case 2, what I'm afraid of is that in Scala `a.infix` can also be written as `a infix` [in a certain condition](https://www.scala-lang.org/api/3.2.1/scala/language$.html#postfixOps-0). 

---

If prior contents matching is such a big concern, it seems a reasonable option to me to remove it and embrace some false positives because it's still better than the status quo. What do you think?

Reply:
What is `notinline`, it looks like a keyword?

Reply:
> embrace some false positives because it's still better than the status quo.

I've often thought if we can't get the highlighting right (say of soft modifiers) that we shouldn't highlight them at all, but there are good arguments for both directions I'd say.

> val transparent = 43

Not sure I understand this example and how you're saying it breaks if you use \b?

Reply:
> What is notinline, it looks like a keyword?

`notinline` is just an arbitrary string that contains `inline`. Probably `\b` will work for this case.

> Not sure I understand this example and how you're saying it breaks if you use \b?

`transparent` is a soft keyword, that means it has special meaning only if it's placed in a certain syntactic position. In this `val transparent = 43` example, `transparent` is just an identifier that is bound to `43`, therefore it should not be highlighted.

Reply:
>  In this `val transparent = 43` example, transparent is just an identifier that is bound to 43, therefore it should not be highlighted.

And since the look-ahead doesn't match there, it would not be highlighted.  So it seems lookahead already handles that case, unless I'm missing something?